include/s3select_functions.h

```diff
@@ -134,11 +135,11 @@ class __function : public base_statement
 private:
   bs_stmt_vec_t arguments;
   std::string name;
-  base_function* m_func_impl;
+  base_function* m_func_impl; // todo: separate the object types used when constructing the __function vs when m_func_impl should be fixed
   s3select_functions* m_s3select_functions;
   variable m_result;
 
-  void _resolve_name()
+  void _resolve_name()  // todo: separate the object types used when constructing the __function vs when m_func_impl should be fixed
```

Comment:
functions are created upon first use (first row) , kind of late-binding. 
thus, it can not be const.
to avoid late-binding, function should be created upon building the AST, it means also to move semantic analysis to earlier stage.
this may happen on later phase, it's quite comprehensive change.